diff --git a/reactor/reactor-grpc-stub/src/main/java/com/salesforce/reactorgrpc/GrpcRetry.java b/reactor/reactor-grpc-stub/src/main/java/com/salesforce/reactorgrpc/GrpcRetry.java
index 249fb0f..479f585 100644
--- a/reactor/reactor-grpc-stub/src/main/java/com/salesforce/reactorgrpc/GrpcRetry.java
+++ b/reactor/reactor-grpc-stub/src/main/java/com/salesforce/reactorgrpc/GrpcRetry.java
@@ -44,7 +44,7 @@ public final class GrpcRetry {
          * @see Flux#retryWhen(Function)
          */
         public static <I, O> Function<? super Mono<I>, Flux<O>> retryWhen(final Function<Mono<I>, Flux<O>> operation, final Function<Flux<Throwable>, ? extends Publisher<?>> whenFactory) {
-            return request -> Flux.defer(() -> operation.apply(request)).retryWhen(whenFactory);
+            return request -> Flux.defer(() -> operation.apply(request)).retryWhen(reactor.util.retry.Retry.withThrowable(whenFactory));
         }

         /**
@@ -92,7 +92,7 @@ public final class GrpcRetry {
          * @see Flux#retryWhen(Function)
          */
         public static <I, O> Function<? super Flux<I>, ? extends Publisher<O>> retryWhen(final Function<Flux<I>, Flux<O>> operation, final Function<Flux<Throwable>, ? extends Publisher<?>> whenFactory) {
-            return request -> Flux.defer(() -> operation.apply(request)).retryWhen(whenFactory);
+            return request -> Flux.defer(() -> operation.apply(request)).retryWhen(reactor.util.retry.Retry.withThrowable(whenFactory));
         }

         /**
@@ -140,7 +140,7 @@ public final class GrpcRetry {
          * @see Flux#retryWhen(Function)
          */
         public static <I, O> Function<? super Flux<I>, Mono<O>> retryWhen(final Function<Flux<I>, Mono<O>> operation, final Function<Flux<Throwable>, ? extends Publisher<?>> whenFactory) {
-            return request -> Mono.defer(() -> operation.apply(request)).retryWhen(whenFactory);
+            return request -> Mono.defer(() -> operation.apply(request)).retryWhen(reactor.util.retry.Retry.withThrowable(whenFactory));
         }

         /**
