diff --git a/java/util/BUILD b/java/util/BUILD
index cfdb28e..43b5276 100644
--- a/java/util/BUILD
+++ b/java/util/BUILD
@@ -5,10 +5,7 @@ java_library(
     srcs = glob([
         "src/main/java/com/google/protobuf/util/*.java",
     ]),
-    javacopts = [
-        "-source 7",
-        "-target 7",
-    ],
+    javacopts = [],
     visibility = ["//visibility:public"],
     deps = [
         "//external:error_prone_annotations",
