diff --git a/private/rules/javadoc.bzl b/private/rules/javadoc.bzl
index 9b90570..bfa5759 100644
--- a/private/rules/javadoc.bzl
+++ b/private/rules/javadoc.bzl
@@ -19,7 +19,7 @@ def _javadoc_impl(ctx):

     jar_file = ctx.actions.declare_file("%s.jar" % ctx.attr.name)

-    classpath = depset(transitive = [dep[JavaInfo].transitive_runtime_jars for dep in ctx.attr.deps])
+    classpath = depset(transitive = [dep[JavaInfo].transitive_compile_time_jars for dep in ctx.attr.deps])

     # javadoc options and javac options overlap, but we cannot
     # necessarily rely on those to derive the javadoc options we need
